main.js

```diff
@@ -1,84 +1,154 @@
-const { app } = require('electron')
-const chokidar = require('chokidar')
-const fs = require('fs')
-const { spawn } = require('child_process')
+const { app } = require("electron");
+const chokidar = require("chokidar");
+const fs = require("fs");
+const { spawn } = require("child_process");
 
-const appPath = app.getAppPath()
-const ignoredPaths = /node_modules|[/\\]\./
-// Main file poses a special case, as its changes are
-// only effective when the process is restarted (hard reset)
-// We assume that electron-reload is required by the main
-// file of the electron application
-const mainFile = module.parent.filename
+const appPath = app.getAppPath();
+const ignoredPaths = /node_modules|[/\\]\./;
 
 /**
  * Creates a callback for hard resets.
  *
- * @param {String} eXecutable path to electron executable
- * @param {String} hardResetMethod method to restart electron
+ * @param {string} eXecutable path to electron executable
+ * @param {string} hardResetMethod method to restart electron
+ * @param {string[]} argv arguments to restart electron with
  * @returns {Function} handler to pass to chokidar
  */
-const createHardresetHandler = (eXecutable, hardResetMethod, argv) =>
-  () => {
-    // Detaching child is useful when in Windows to let child
-    // live after the parent is killed
-    const args = (argv || []).concat([appPath])
-    const child = spawn(eXecutable, args, {
-      detached: true,
-      stdio: 'inherit'
-    })
-    child.unref()
-    // Kamikaze!
+const createHardresetHandler = (eXecutable, hardResetMethod, argv) => () => {
+  // Detaching child is useful when in Windows to let child
+  // live after the parent is killed
+  const args = [appPath].concat(argv || []);
```

Comment:
This does fix #87, but it seems to removes the ability to specify arguments to electron itself. To preserve that a second argument lists should be provided in options, something like `electronArgv`, and also passed to this function. Then you would be able to do something like:
```
const args = (electronArgv || []);
args.push(appPath);
args.push(...argv || []);
```

Reply:
added `appArgv` & `electronArgv` options